code/modules/clothing/clothing.dm

```diff
@@ -374,6 +376,24 @@ BLIND     // can't see anything
 		if(bloody)
 			. += image("icon"='icons/effects/blood.dmi', "icon_state"="shoeblood")
 
+/obj/item/clothing/shoes/equipped(mob/user, slot)
+	. = ..()
+	if(offset && slot_flags & slotdefine2slotbit(slot))
+		user.pixel_y += offset
+		worn_y_dimension -= (offset * 2)
+		user.update_inv_shoes()
+		dropped_twice = TRUE
+
+/obj/item/clothing/shoes/proc/restore_offsets(mob/user)
+	dropped_twice = FALSE
+	user.pixel_y -= offset
+	worn_y_dimension = 32

```

Comment:
I just have a feeling that at some point something's going to go wrong with this. And it'll be hilarious.